### PR TITLE
Introduce a centralised way to emit telemetry from calls of injectable functions

### PR DESCRIPTION
**Pull request by Iku-turso**
The PR also includes the first actual source of telemetry: navigate-to-route.injectable.ts. This means that all (or most) navigations now emit telemetry.

Notable: in scope of this PR, telemetry is emitted to an event bus, for which further processing is done somewhere else.

Notable2: this PR bumps `injectable` for its own purposes. An indirectly related byproduct of this is that now calls to `di.register()` require an explicit MobX transaction, eg. `runInAction(() => { di.register(...); });`.

Notable3: this PR uses `decorable: false` as a blunt object to prevent recursive decoration of decorators. A better solution is in the horizon.

This is a new requirement from `injectable-extension-for-mobx`, as otherwise registrations could surprisingly cause untimely, and too numerous observations by `computedInjectMany`. The bad thing is the duplication, but I do currently consider this as favorable, as now a dev will always know that registrations of injectables with tokens may cause triggering of observations.

### CODE REVIEW

File: src/features/telemetry/renderer/telemetry-decorator.injectable.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    };
+  },
+
+  decorable: false,
+  // Todo: this is required because of imperfect typing in injectable.
+  lifecycle: lifecycleEnum.singleton,
+  injectionToken: instantiationDecoratorToken,
+});
+
+const shouldEmitTelemetry = (currentContext: any, whiteList: any) => (
```
Comment (jweak) [inline src/features/telemetry/renderer/telemetry-decorator.injectable.ts]: We can probably make this whiteList at least an Array type if not string[]?

Comment (Iku-turso) [reply src/features/telemetry/renderer/telemetry-decorator.injectable.ts]: True, one sec.

File: src/features/telemetry/renderer/telemetry-white-list-for-functions.injectable.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+
+const telemetryWhiteListForFunctionsInjectable = getInjectable({
+  id: "telemetry-white-list-for-functions",
+  instantiate: () => ["some-placeholder-injectable-id"],
```
Comment (jweak) [inline src/features/telemetry/renderer/telemetry-white-list-for-functions.injectable.ts:9]: Do we need this placeholder?

File: src/features/telemetry/renderer/telemetry-decorator.injectable.ts (hunk 1 above)
Comment (Iku-turso) [reply src/features/telemetry/renderer/telemetry-decorator.injectable.ts]: Fixed, also simplified the related code a bit.

File: src/features/telemetry/renderer/telemetry-white-list-for-functions.injectable.ts (hunk 2 above)
Comment (Iku-turso) [reply src/features/telemetry/renderer/telemetry-white-list-for-functions.injectable.ts:9]: It's there for documentation and inferred typing. I'd say no harm?

Comment (jweak) [reply src/features/telemetry/renderer/telemetry-white-list-for-functions.injectable.ts:9]: Yeah, no harm.

**Review by jweak — APPROVED** [review]

File: src/common/app-event-bus/app-event-bus.injectable.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ const appEventBusInjectable = getInjectable({
   id: "app-event-bus",
   instantiate: () => appEventBus,
   causesSideEffects: true,
+  decorable: false,
```
Comment (Nokel81) [inline src/common/app-event-bus/app-event-bus.injectable.ts:12]: I still don't quite understand what this means, we don't use decorators on `getInjectable` anyway...

File: src/common/catalog-entities/__tests__/kubernetes-cluster.test.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ describe("kubernetesClusterCategory", () => {
   let kubernetesClusterCategory: KubernetesClusterCategory;
 
   beforeEach(() => {
-    const di = getDiForUnitTesting();
+    const di = getDiForUnitTesting({ doGeneralOverrides: true });
```
Comment (Nokel81) [inline src/common/catalog-entities/__tests__/kubernetes-cluster.test.ts:15]: Can we just make this the default now anyway (or remove the option entirely)?

File: src/features/cluster/workloads/overview/extension-api/order-of-workload-overview-details.test.tsx [hunk 5]
```diff
@@ -20,11 +20,13 @@ describe("order of workload overview details", () => {
...
-        someCoreItemWithLowOrderNumberInjectable,
-        someCoreItemWithHighOrderNumberInjectable,
-        someCoreItemWithDefaultOrderNumberInjectable,
-      );
+      runInAction(() => {
+        windowDi.register(
+          someCoreItemWithLowOrderNumberInjectable,
+          someCoreItemWithHighOrderNumberInjectable,
+          someCoreItemWithDefaultOrderNumberInjectable,
+        );
```
Comment (Nokel81) [inline src/features/cluster/workloads/overview/extension-api/order-of-workload-overview-details.test.tsx:28]: Given that `.register` already takes multiple items. Wouldn't it just make sense to mark it as `action` instead?

File: src/features/quitting-and-restarting-the-app/opening-application-window-using-tray.test.ts [hunk 6]
```diff
@@ -61,10 +62,12 @@ describe("opening application window using tray", () => {
           return browserWindow;
         });
 
-        (mainDi as any).decorateFunction(
-          createElectronWindowInjectable,
-          createElectronWindowMock,
-        );
+        runInAction(() => {
+          (mainDi as any).decorateFunction(
```
Comment (Nokel81) [inline src/features/quitting-and-restarting-the-app/opening-application-window-using-tray.test.ts:66]: Given that we just updated to a new major version, shouldn't `decorateFunction` be exported now?

Comment (Nokel81) [reply src/features/quitting-and-restarting-the-app/opening-application-window-using-tray.test.ts:66]: Also, can we get some documentation about what this does?

**Review by Nokel81 — APPROVED** [review]

**Pull request merged**